bitarray/sparse_bitarray.go

```diff
@@ -296,6 +300,67 @@ func (sba *sparseBitArray) IntersectsBetween(other BitArray, start, stop uint64)
 	return true
 }
 
+// WriteSparse serializes the sparseBitArray and passes it to the writer.
+func WriteSparse(w io.Writer, ba *sparseBitArray) error {
```

Comment:
Yeah, you could just put (un)marhsal methods on the interface that take the io interfaces, or you could make (un)marshal functions on the package that take the BitArray interface and return byte arrays.  The latter is probably closer to how the standard library marshaling functionality works.